psse/psse-model/src/main/java/com/powsybl/psse/model/io/AbstractRecordGroup.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.psse.model.io;
+
+import com.powsybl.psse.model.PsseException;
+import com.powsybl.psse.model.PsseVersion;
+import com.univocity.parsers.common.processor.BeanListProcessor;
+import com.univocity.parsers.common.processor.BeanWriterProcessor;
+import com.univocity.parsers.csv.CsvParser;
+import com.univocity.parsers.csv.CsvParserSettings;
+import com.univocity.parsers.csv.CsvWriter;
+import com.univocity.parsers.csv.CsvWriterSettings;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.*;
+
+import static com.powsybl.psse.model.io.FileFormat.JSON;
+import static com.powsybl.psse.model.io.FileFormat.LEGACY_TEXT;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+public abstract class AbstractRecordGroup<T> {
+
+    protected final RecordGroupIdentification identification;
+    private final String[] fieldNames;
+    private final Map<PsseVersion.Major, String[]> fieldNamesByVersionMajor = new EnumMap<>(PsseVersion.Major.class);
+    private String[] quotedFields;
+    private final Map<FileFormat, RecordGroupReaderWriter<T>> readersWriters = new EnumMap<>(FileFormat.class);
```

Comment:
I wonder if we should split this interface in two different interfaces. I think the implementation will result in huge files

Reply:
I am not sure we have to split readers and writers. The implementation of `read`/`write` methods are quite compact, and seeing them together in the same class helps understanding the processing.